src/vm/gdbjit.cpp

```diff
@@ -2068,29 +2069,38 @@ bool NotifyGdb::BuildDebugPub(MemBuf& buf, const char* name, uint32_t size, uint
 /* Store addresses and names of the called methods into symbol table */
 bool NotifyGdb::CollectCalledMethods(CalledMethod* pCalledMethods)
 {
-    int calledCount = 0;
+    AddrSet tmpCodeAddrs;
+
     CalledMethod* pList = pCalledMethods;
 
      /* count called methods */
     while (pList != NULL)
     {
-        calledCount++;
+        TADDR callAddr = (TADDR)pList->GetCallAddr();
+        if (!tmpCodeAddrs.Contains(callAddr) && !codeAddrs.Contains(callAddr)) {
+            tmpCodeAddrs.Add(callAddr);
+        }
         pList = pList->GetNext();
     }
 
-    SymbolCount = 1 + method_count + calledCount;
+    SymbolCount = 1 + method_count + tmpCodeAddrs.GetCount();
     SymbolNames = new (nothrow) Elf_Symbol[SymbolCount];
 
     pList = pCalledMethods;
-    for (int i = 1 + method_count; i < SymbolCount; ++i)
+    for (int i = 1 + method_count; i < SymbolCount;)
     {
-        char buf[256];
-        MethodDesc* pMD = pList->GetMethodDesc();
-        sprintf(buf, "__thunk_%s", pMD->GetName());
-        SymbolNames[i].m_name = new char[strlen(buf) + 1];
-        strcpy((char*)SymbolNames[i].m_name, buf);
         TADDR callAddr = (TADDR)pList->GetCallAddr();
-        SymbolNames[i].m_value = callAddr;
+        if (!codeAddrs.Contains(callAddr))
+        {
+            char buf[256];
+            MethodDesc* pMD = pList->GetMethodDesc();
+            sprintf(buf, "__thunk_%s", pMD->GetName());
```

Comment:
This is dangerous, you can easily overrun the buffer for longer symbols. Since you are allocating a buffer for it later anyways, can you please allocate properly sized buffer  right away and sprintf into it?